src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -77,10 +89,46 @@ public List<RawEntityTypeSummary> getEntityTypeSummary(Set<UUID> entityTypeIds)
       );
   }
 
+  public List<EntityTypeColumn> fetchColumnNamesForCustomFields(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getEntityTypeDefinition(entityTypeId)
+      .map(EntityType::getSourceView)
+      .orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    return jooqContext
+      .select(field(REQUIRED_FIELD_NAME), field(REF_ID))
+      .from(sourceViewName)
+      .where(field(WHERE_CLAUSE).eq(CUSTOM_FIELD_TYPE))
```

Comment:
This where condition will be removed when we will handle other CustomFieldTypes. We can create a generic method with if and else statement. Right now if we do so we need to "null" in the else part

Reply:
Could you add a comment here about that? It could be a pretty subtle thing to track down later, so a big sign pointing at this line, saying "HEY!!! YOU'RE ADDING MORE CUSTOM FIELD TYPES? COOL! PAY ATTENTION TO THIS!!!!" could be invaluable later on if there's any real delay between this PR and when we do more types